common/perf-test-core/src/main/java/com/azure/perf/test/core/MockEventProcessor.java

```diff
@@ -0,0 +1,150 @@
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.concurrent.locks.ReentrantLock;
+import java.util.function.Consumer;
+import java.util.stream.IntStream;
+
+/**
+ * Represents a Mock event processor.
+ */
+public class MockEventProcessor {
+    private final Consumer<MockErrorContext> processError;
+    private final Consumer<MockEventContext> processEvent;
+    private boolean process;
+    private final double maxEventsPerSecondPerPartition;
+    private final int maxEventsPerSecond;
+    private final int partitions;
+    private final Duration errorAfter;
+    private boolean errorRaised;
+    private final ReentrantLock errorLock;
+
+    private final MockEventContext[] mockEventContexts;
+    private int[] eventsRaised;
+    private long startTime;
+
+    private final AtomicReference<ScheduledFuture<?>> runner = new AtomicReference<>();
+    private final AtomicReference<ScheduledExecutorService> scheduler = new AtomicReference<>();
+
+    /**
+     * Creates an instance of a mock event processor
+     *
+     * @param partitions the number of partitions
+     * @param maxEventsPerSecond the maximum events per second to send, optional.
+     * @param errorAfter the duration after which processor should error out, optional.
+     * @param processError the consumer to process the error.
+     * @param processEvent the consumer to process the event.
+     */
+    public MockEventProcessor(int partitions, int maxEventsPerSecond, Duration errorAfter,
+                              Consumer<MockErrorContext> processError, Consumer<MockEventContext> processEvent) {
+        this.processError = processError;
+        this.processEvent = processEvent;
+        this.partitions = partitions;
+        this.maxEventsPerSecond = maxEventsPerSecond;
+        this.maxEventsPerSecondPerPartition = ((double) maxEventsPerSecond) / partitions;
+        this.errorAfter = errorAfter;
+        this.errorLock = new ReentrantLock();
+
+        mockEventContexts = new MockEventContext[partitions];
+        IntStream.range(0, partitions).boxed().forEach(integer -> {
+            mockEventContexts[integer] = new MockEventContext(integer, "Hello");
+        });
+        this.eventsRaised = new int[partitions];
+    }
+
+    /**
+     * Starts the event processor.
+     */
+    public synchronized void start() {
+        eventsRaised = new int[eventsRaised.length];
+        process = true;
+        errorRaised = false;
+        startTime = System.nanoTime();
+        ScheduledExecutorService executor = Executors.newSingleThreadScheduledExecutor();
+        scheduler.set(executor);
+
+        Double jitterInMillis =
+            ThreadLocalRandom.current().nextDouble() * TimeUnit.SECONDS.toMillis(0);
+
+        runner.set(scheduler.get().schedule(this::processEvents,
```

Comment:
We want to guarantee the following invariants:

1. Within a partition, events are raised sequentially.
2. Across all partitions, events are raised in parallel.

Does the Java code currently in this class implement this?  I'm not sure.  The code in `processEvents()` looks like each partition is processed sequentially rather than in parallel.  Combined with the code in `process(partition)`, I event think all the events might be sent to the first partition with none sent to later partitions.

Reply:
Updated the code.
But I believe, Java Event Processors, operate syncrhronously.

Reply:
Discussed this with @g2vinay earlier and we do process events parallelly. So, we need to have support for what @mikeharder  is suggesting above.

Reply:
I've added the parallel logic now, in processEvents function.